flake.nix

```
@@ -0,0 +1,58 @@
+{
+  description = "Next generation distributed, event-driven, parallel config management!";
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:urandom2/nixpkgs/mgmt";
+  };
+  outputs = {
+    flake-utils,
+    nixpkgs,
+    self,
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      buildInputs = with pkgs; [
+        augeas
+        libvirt
+        libxml2
+      ];
+      nativeBuildInputs = with pkgs; [
+        gotools
+        nex
+        pkg-config
+        ragel
+      ];
+      pkgs = nixpkgs.legacyPackages.${system};
+    in {
+      devShells.default = pkgs.mkShell {
+        inherit buildInputs nativeBuildInputs;
+        packages = [pkgs.go];
+      };
+      packages.default = let
+        pname = "mgmt";
+        version = "0.0.22";
```

unfortunately, flakes cannot consume version strings from vcs, so you will be required to manually update the string before releases, or just leave the flake as something like `unstable-20221107155054` like so:
```suggestion
        version = "unstable-${self.lastModifiedDate}";
```